drivers/unix/os_unix.cpp

```diff
@@ -246,10 +253,29 @@ void OS_Unix::delay_usec(uint32_t p_usec) const {
 }
 uint64_t OS_Unix::get_ticks_usec() const {
 
-	struct timeval tv_now;
-	gettimeofday(&tv_now, NULL);
+#if defined(__APPLE__)
+	static double scale = 0;
+	if (scale == 0 && ticks_start == 0) {
+		mach_timebase_info_data_t info;
+		kern_return_t ret = mach_timebase_info(&info);
+		if (ret != 0) {
+			// Only print once (return 1 so we will not enter the if).
+			ERR_EXPLAIN("CLOCK IS NOT WORKING!");
+			ERR_FAIL_V(1);
+		}
+		scale = (double)info.numer / (double)info.denom;
+	}
 
-	uint64_t longtime = (uint64_t)tv_now.tv_usec + (uint64_t)tv_now.tv_sec * 1000000L;
+	uint64_t longtime = mach_absolute_time() * scale;
+#else
+	struct timespec tv_now = { 0, 0 };
+	if (clock_gettime(GODOT_CLOCK, &tv_now) != 0 && ticks_start == 0) {
```

Comment:
I wonder if it would be better to use directly:
```
ERR_EXPLAIN("CLOCK IS NOT WORKING!");
ERR_FAIL_COND_V(clock_gettime(GODOT_CLOCK, &tv_now) != 0 && ticks_start == 0, 1);
```
to benefit from the `unlikely()` in `ERR_FAIL_COND*`? (CC @hpvb)

It's arguably less pretty but since this is called in the game loop, might be worth the optimization.

Reply:
That said allocating memory for a string in `ERR_EXPLAIN` might cost much than the lack of usage of `unlikely()`, so nevermind my comment :P

Reply:
Yeah, actually, that was my main concern about it. But there might still be a better solution...